modin/experimental/core/execution/unidist/implementations/pandas_on_unidist/io/sql.py

```diff
@@ -0,0 +1,275 @@
+    str
+        Query string.
+    """
+    return "SELECT * FROM {0}".format(name)
+
+
+def check_query(query):
+    """
+    Check query sanity.
+
+    Parameters
+    ----------
+    query : str
+        Query string.
+    """
+    q = query.lower()
+    if "select " not in q:
+        raise InvalidQuery("SELECT word not found in the query: {0}".format(query))
+    if " from " not in q:
+        raise InvalidQuery("FROM word not found in the query: {0}".format(query))
+
+
+def get_query_columns(engine, query):
+    """
+    Extract columns names and python types from the `query`.
+
+    Parameters
+    ----------
+    engine : sqlalchemy.engine.base.Engine
+        SQLAlchemy connection engine.
+    query : str
+        SQL query.
+
+    Returns
+    -------
+    OrderedDict
+        Dictionary with columns names and python types.
+    """
+    con = engine.connect()
+    result = con.execute(query).fetchone()
+    values = list(result)
+    cols_names = list(result.keys())
+    cols = OrderedDict()
+    for i in range(len(cols_names)):
+        cols[cols_names[i]] = type(values[i]).__name__
+    return cols
+
+
+def check_partition_column(partition_column, cols):
+    """
+    Check `partition_column` existence and it's type.
+
+    Parameters
+    ----------
+    partition_column : str
+        Column name used for data partitioning between the workers.
+    cols : OrderedDict/dict
+        Dictionary with columns names and python types.
+    """
+    for k, v in cols.items():
+        if k == partition_column:
+            if v == "int":
+                return
+            else:
+                raise InvalidPartitionColumn(
+                    "partition_column must be int, and not {0}".format(v)
+                )
+    raise InvalidPartitionColumn(
+        "partition_column {0} not found in the query".format(partition_column)
+    )
+
+
+def get_query_info(sql, con, partition_column):
+    """
+    Compute metadata needed for query distribution.
+
+    Parameters
+    ----------
+    sql : str
+        SQL query to be executed or a table name.
+    con : SQLAlchemy connectable or str
+        Database connection or url string.
+    partition_column : str
+        Column name used for data partitioning between the workers.
+
+    Returns
+    -------
+    list
+        Columns names list.
+    str
+        Query string.
+    """
+    engine = create_engine(con)
+    if is_table(engine, sql):
+        table_metadata = get_table_metadata(engine, sql)
+        query = build_query_from_table(sql)
+        cols = get_table_columns(table_metadata)
+    else:
+        check_query(sql)
+        query = sql.replace(";", "")
+        cols = get_query_columns(engine, query)
+    # TODO allow validation that takes into account edge cases of pandas e.g. "[index]"
+    # check_partition_column(partition_column, cols)
+    # TODO partition_column isn't used; we need to use it;
+    cols_names = list(cols.keys())
+    return cols_names, query
```

Comment:
```suggestion
    return list(cols.keys()), query
```
I see no use for that extra variable

Reply:
let's leave this comment to be fixed in https://github.com/modin-project/modin/issues/5275, in order to simplify code deduplication.